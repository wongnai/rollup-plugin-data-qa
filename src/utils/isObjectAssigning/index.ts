export default function isObjectAssigning(node: Record<string, any>) {
	return node.callee?.object?.name === 'Object' && node.callee?.property?.name === 'assign'
}
