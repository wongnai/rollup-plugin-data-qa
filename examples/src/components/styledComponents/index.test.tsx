import { render, screen, within } from '../../utils/testUtils'
import Component01 from './Component01'
import Component02 from './Component02'
import Component03 from './Component03'
import Component04 from './Component04'
import Component05 from './Component05'
import Component06 from './Component06'
import Component07 from './Component07'
import Component08 from './Component08'

describe('styled components', () => {
	it('01 - styled component in a div', () => {
		const { container } = render(<Component01 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('styled-a')).toBeVisible()
		expect(screen.getByDataQa('component01')).toBeVisible()
	})

	it('02 - returning styled component', () => {
		const { container } = render(<Component02 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('component02')).toBeVisible()
		expect(screen.queryByDataQa('styled-a')).not.toBeInTheDocument()
	})

	it('03 - returning styled component but with useMemo above', () => {
		const { container } = render(<Component03 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('component03')).toBeVisible()
		expect(screen.getByDataQa('b')).toBeVisible()
		expect(screen.queryByDataQa('styled-a')).not.toBeInTheDocument()
	})

	it('04 - component with attrs', () => {
		const { container } = render(<Component04 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('styled-a')).toBeVisible()
	})

	it('05 - styled styled component', () => {
		const { container } = render(<Component05 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('component05')).toBeVisible()
		expect(screen.getAllByDataQa('styled-b')).toHaveLength(2)
	})

	it('06 - styled styled component with attrs', () => {
		const { container } = render(<Component06 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.queryByDataQa('component06')).not.toBeInTheDocument()
		expect(screen.getAllByDataQa('styled-a')).toHaveLength(2)
		expect(screen.getByDataQa('styled-b')).toBeVisible()
	})

	it('07 - styled with props function in attrs', () => {
		const { container } = render(<Component07 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('component07')).toBeVisible()
		expect(within(screen.getByDataQa('component07')).getByDataQa('styled-a')).toBeVisible()
	})

	it('08 - component with custom hook that returns object with styled component', () => {
		const { container } = render(<Component08 />)

		expect(container.firstChild).toMatchSnapshot()
		expect(screen.getByDataQa('component08')).toBeVisible()
		expect(screen.getByDataQa('icon-default')).toBeVisible()
	})
})
