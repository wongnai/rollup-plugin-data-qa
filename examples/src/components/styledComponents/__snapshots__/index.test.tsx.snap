// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`styled components > 01 - styled component in a div 1`] = `
.c0 {
  color: red;
}

<div
  data-qa="component01"
>
  <div
    class="c0"
    data-qa="styled-a"
  >
    xxx
  </div>
</div>
`;

exports[`styled components > 02 - returning styled component 1`] = `
.c0 {
  color: red;
}

<div
  class="c0"
  data-qa="component02"
>
  xxx
</div>
`;

exports[`styled components > 03 - returning styled component but with useMemo above 1`] = `
.c0 {
  color: red;
}

<div
  class="c0"
  data-qa="styled-a"
>
  <div
    data-qa="component03"
  >
    xxx
  </div>
</div>
`;

exports[`styled components > 04 - component with attrs 1`] = `
.c0 {
  color: red;
}

<div
  class="c0 a"
  data-qa="styled-a"
>
  xxx
</div>
`;

exports[`styled components > 05 - styled styled component 1`] = `
.c0 {
  color: blue;
}

.c1 {
  color: red;
}

<div
  class="c0 c1"
  data-qa="component05"
>
  <div
    class="c0 c1"
    data-qa="styled-b"
  >
    aaa
  </div>
  <div
    class="c0"
    data-qa="styled-b"
  >
    bbb
  </div>
</div>
`;

exports[`styled components > 06 - styled styled component with attrs 1`] = `
.c0 {
  color: blue;
}

.c1 {
  color: red;
}

<div
  class="c0 c1 b a"
  data-qa="styled-a"
>
  <div
    class="c0 c1 b a"
    data-qa="styled-a"
  >
    xxx
  </div>
  <div
    class="c0 b"
    data-qa="styled-b"
  >
    xxx
  </div>
</div>
`;

exports[`styled components > 07 - styled with props function in attrs 1`] = `
.c0 {
  color: red;
}

<div
  class="c0 a"
  data-qa="component07"
>
  <div
    class="c0 a"
    data-qa="styled-a"
  >
    xxx
  </div>
</div>
`;

exports[`styled components > 08 - component with custom hook that returns object with styled component 1`] = `
.c0 {
  color: green;
}

<div
  class="default"
  data-qa="component08"
>
  <div>
    <svg
      class="c0"
      data-qa="icon-default"
    />
  </div>
  <div>
    labelDefault
  </div>
</div>
`;
